x-pack/plugins/task_manager/server/monitoring/configuration_statistics.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { combineLatest, of } from 'rxjs';
+import { pick, merge } from 'lodash';
+import { map, startWith } from 'rxjs/operators';
+import { AggregatedStatProvider } from './runtime_statistics_aggregator';
+import { TaskManagerConfig } from '../config';
+import { ManagedConfiguration } from '../lib/create_managed_configuration';
+
+const CONFIG_FIELDS_TO_EXPOSE = [
+  'request_capacity',
+  'max_poll_inactivity_cycles',
+  'monitored_aggregated_stats_refresh_rate',
+  'monitored_stats_running_average_window',
+  'monitored_task_execution_thresholds',
+] as const;
+
+export type ConfigStat = Pick<
+  TaskManagerConfig,
+  'max_workers' | 'poll_interval' | typeof CONFIG_FIELDS_TO_EXPOSE[number]
+>;
+
+export function createConfigurationAggregator(
+  config: TaskManagerConfig,
+  managedConfig: ManagedConfiguration
+): AggregatedStatProvider<ConfigStat> {
+  return combineLatest([
+    of(pick(config, ...CONFIG_FIELDS_TO_EXPOSE)),
+    managedConfig.pollIntervalConfiguration$.pipe(
+      startWith(config.poll_interval),
```

Comment:
Is `startWith` still needed here? I was thinking the one [from managed configuration](https://github.com/elastic/kibana/blob/master/x-pack/plugins/task_manager/server/lib/create_managed_configuration.ts#L49) would endup here.

Reply:
As `combineLatest` stalls if any single stream is missing a value, it's a good practice to ensure you have at least an initial value for all streams (unless that breaks the logic somehow).
In our case, doing this ensures that the first value of the `configuration` stat is the actual initial value of the TaskManagerConfig, which feels more correct anyway.

This also ensures that we don't _stall_ until the managed config emit another config and as the managed config emits in response to errors, it would be possible to accidentally stall here if no error happen _after_ we subscribe to these streams. (Keep in mind the initial values of the managed config would not be streamed into the monitored stats because you only get the values _after_ you subscribe, not the values emitted before you subscribe).